lib/rucio/transfertool/cms_fts3.py

```diff
@@ -0,0 +1,358 @@
+        expected_transfer_id = None
+        if self.deterministic:
+            job_params = job_params.copy()
+            job_params["id_generator"] = "deterministic"
+            job_params["sid"] = files[0]['metadata']['request_id']
+            expected_transfer_id = self.__get_deterministic_id(job_params["sid"])
+            logging.debug("Submit bulk transfers in deterministic mode, sid %s, expected transfer id: %s", job_params["sid"], expected_transfer_id)
+
+        # bulk submission
+        jobID = None
+        transfers = []
+        for _file in files:
+            for source, destination in zip(_file["sources"], _file["destinations"]):
+                transfers.append(fts.new_transfer(source, destination,
+                                                  activity=_file['activity'],
+                                                  metadata=_file['metadata'],
+                                                  filesize=_file['filesize'],
+                                                  checksum=_file['checksum']))
+
+        try:
+            context = fts.Context(self.external_host,
+                                  ucert=usercert,
+                                  ukey=usercert,
+                                  verify=True,
+                                  capath='/etc/grid-security/certificates/')
+
+            job = fts.new_job(transfers,
+                              overwrite=job_params['overwrite'],
+                              verify_checksum=job_params['verify_checksum'],
+                              metadata=job_params['job_metadata'],
+                              copy_pin_lifetime=job_params['copy_pin_lifetime'],
+                              bring_online=job_params['bring_online'],
+                              source_spacetoken=None,
+                              spacetoken=None,
+                              priority=job_params['priority'])
+
+            # submission for bindings checks the delegation, so the usual delegation part is useless here
+            jobID = fts.submit(context, job, delegation_lifetime=timedelta(hours=12), delegate_when_lifetime_lt=timedelta(hours=8))
+        except ServerError:
+            logging.error("Server side exception during FTS job submission.")
+            return None
+        except ClientError:
+            logging.error("Client side exception during FTS job submission.")
+            return None
+        except BadEndpoint:
+            logging.error("Wrong FTS endpoint: %s", self.external_host)
+            return None
+
+        if jobID:
+            record_counter('transfertool.fts3.%s.submission.success' % self.__extract_host(self.external_host), len(files))
+            transfer_id = jobID
+        else:
+            transfer_id = jobID
+            logging.error("Unexpected failure during FTS job submission.")
+            record_counter('transfertool.fts3.%s.submission.failure' % self.__extract_host(self.external_host), len(files))
```

Comment:
a separate subcategory for these might be useful, e.g., `transfertool.fts3myproxy.%s.submission` ?

Reply:
ok, I agree, no problem